Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/JointLimitOptimizer.cpp

```diff
@@ -0,0 +1,149 @@
+            GetBoneDirection(physicsSetupManipulatorData.m_actor->GetSkeleton(), physicsSetupManipulatorData.m_node);
+        const Pose* bindPose = physicsSetupManipulatorData.m_actor->GetSkeleton()->GetBindPose();
+        const Transform& nodeBindTransform = bindPose->GetModelSpaceTransform(physicsSetupManipulatorData.m_node->GetNodeIndex());
+        const Transform& parentBindTransform = physicsSetupManipulatorData.m_node->GetParentNode()
+            ? bindPose->GetModelSpaceTransform(physicsSetupManipulatorData.m_node->GetParentIndex())
+            : Transform::CreateIdentity();
+        const AZ::Quaternion& nodeBindRotationWorld = nodeBindTransform.m_rotation;
+        const AZ::Quaternion& parentBindRotationWorld = parentBindTransform.m_rotation;
+        AZStd::unique_ptr<AzPhysics::JointConfiguration> jointInitialConfiguration =
+            jointHelpersInterface->ComputeInitialJointLimitConfiguration(
+                physicsSetupManipulatorData.m_jointConfiguration->RTTI_GetType(),
+                parentBindRotationWorld,
+                nodeBindRotationWorld,
+                boneDirection,
+                {});
+
+        if (!jointInitialConfiguration)
+        {
+            AZ_Warning("Joint Limit Optimizer", false, "Cannot optimize joint limit as computing initial configuration failed.");
+            return;
+        }
+
+        // check how many motions are available to help decide how many samples to take per motion
+        MotionManager& motionManager = EMotionFX::GetMotionManager();
+        const size_t numMotionSets = motionManager.GetNumMotionSets();
+        int numMotions = 0;
+        for (size_t motionSetIndex = 0; motionSetIndex < numMotionSets; motionSetIndex++)
+        {
+            const MotionSet* motionSet = motionManager.GetMotionSet(motionSetIndex);
+            const MotionSet::MotionEntries& motionEntries = motionSet->GetMotionEntries();
+            for (const auto& motionEntryPair : motionEntries)
+            {
+                MotionSet::MotionEntry* motionEntry = motionEntryPair.second;
+                motionSet->LoadMotion(motionEntry);
+                const Motion* motion = motionEntry->GetMotion();
+                if (!motion || !motion->GetMotionData())
+                {
+                    continue;
+                }
+                const AZ::Outcome<size_t> jointIndexOutcome =
+                    motion->GetMotionData()->FindJointIndexByName(physicsSetupManipulatorData.m_node->GetNameString());
+                numMotions += jointIndexOutcome.IsSuccess();
+            }
+        }
+
+        if (numMotions == 0)
+        {
+            AZ_Warning("Joint Limit Optimizer", false, "Cannot optimize joint limit because no motions were found.");
+            return;
+        }
+
+        // now do the actual sampling of rotation values
+        const int numSamplesPerMotion = AZ::GetMin(JointLimitOptimizerMaxSamplesPerMotion, JointLimitOptimizerTotalSamples / numMotions);
```

Comment:
As we're iterating over the motions previously anyway, we could just get the actual number of samples here - why these approximations?